platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/pipeline/VirtualPipeline.java

```diff
@@ -544,16 +558,13 @@ private void hashFlushMerge() {
             if (!copy.isImmutable()) {
                 break;
             }
-            boolean flushed = false;
+            compact(copy);
```

Comment:
Here is the core of the fix. Every copy, even if not the latest one, is first compacted if needed (if its estimated size exceeds the threshold). Then it can be either flushed (if the size is still more than the threshold after compaction, and the copy is the oldest) or merged (if the size is under threshold), or nothing (if the size exceeds the threshold, but the copy is not the oldest).